OData/src/System.Web.OData/OData/Formatter/Serialization/ODataPrimitiveSerializer.cs

```diff
@@ -144,14 +144,26 @@ internal static object ConvertUnsupportedPrimitives(object value)
 
                     case TypeCode.DateTime:
                         DateTime dateTime = (DateTime)value;
-                        TimeZoneInfo timeZone = TimeZoneInfoHelper.TimeZone;
-                        if (dateTime.Kind == DateTimeKind.Utc || dateTime.Kind == DateTimeKind.Local)
+
+                        if (dateTime == DateTime.MinValue)
```

Comment:
For example, if `dateTime` = '0001-01-01 01:00:00', but the `timeZone` is GMT+8; then the line 165, 166 might throw out of range exception.